include/rcpputils/filesystem_helper.hpp

```diff
@@ -412,6 +412,35 @@ inline path temp_directory_path()
   return path(temp_path);
 }
 
+/**
+ * \brief Return current working directory.
+ *
+ * \return The current working directory.
+ *
+ * \throws std::system_error
+ */
+inline path current_path()
+{
+  char * cwd;
+#ifdef _WIN32
+#ifdef UNICODE
+#error "rcpputils::fs does not support Unicode paths"
+#endif
+  cwd = _getcwd(NULL, 0);
+#else
+  cwd = getcwd(NULL, 0);
+#endif
+  if (nullptr == cwd) {
+    std::error_code ec{errno, std::system_category()};
+    errno = 0;
```

Comment:
@cottsay I'd rater not change `errno` in the process.

Reply:
I'm following the pattern that exists in this file already. If you think that pattern should change, you should start that discussion, but I think that consistent behavior is more important for now.
https://github.com/ros2/rcpputils/blob/eee6c97cbed514afc0ed1160f30a0eb176fd8a0b/include/rcpputils/filesystem_helper.hpp#L197-L199